pamu2fcfg/pamu2fcfg.c

```diff
@@ -203,24 +221,73 @@ static int make_cred(const struct args *args, const char *path, fido_dev_t *dev,
        r == FIDO_ERR_PIN_BLOCKED)) {
     n = snprintf(prompt, sizeof(prompt), "Enter PIN for %s: ", path);
     if (n < 0 || (size_t) n >= sizeof(prompt)) {
-      fprintf(stderr, "error: snprintf prompt");
-      return -1;
+      fprintf(stderr, "error: snprintf prompt\n");
+      goto err;
     }
-    if (!readpassphrase(prompt, pin, sizeof(pin), RPP_ECHO_OFF)) {
-      fprintf(stderr, "error: failed to read pin");
-      explicit_bzero(pin, sizeof(pin));
-      return -1;
+    if ((pin = malloc(BUFSIZE)) == NULL) {
+      fprintf(stderr, "error: malloc\n");
+      goto err;
+    }
+    if (!readpassphrase(prompt, pin, BUFSIZE, RPP_ECHO_OFF)) {
+      fprintf(stderr, "error: failed to read pin\n");
+      goto err;
     }
     r = fido_dev_make_cred(dev, cred, pin);
   }
-  explicit_bzero(pin, sizeof(pin));
 
   if (r != FIDO_OK) {
     fprintf(stderr, "error: fido_dev_make_cred (%d) %s\n", r, fido_strerr(r));
-    return -1;
+    goto err;
   }
 
-  return 0;
+  if (args->authtok) {
+    if (!readpassphrase("Enter auth token: ", authtok, sizeof(authtok),
+                        RPP_ECHO_OFF)) {
+      fprintf(stderr, "error: failed to read auth token\n");
+      goto err;
+    }
+
+    if (args->pam_userpresence == 0 && args->no_user_presence) {
+      fprintf(stderr, "error: user presence is required for auth token\n");
+      goto err;
+    }
```

Comment:
We can simply make `--authtok` comflict with `--no-user-presence` in pamu2fcfg.

Reply:
Yes, my concern is the user editing their credentials at some later point in time, removing the `+presence` option but not the encrypted authtok. They'd then be locked out of their account since the authenticator will refuse to retrieve an assertion with the HMAC secret extension enabled.

Reply:
I see what you mean now. Changed so that `userpresence=0` and no `+presence` implies no `allowauthtok`.